client/src/main/java/org/apache/celeborn/client/read/DfsPartitionReader.java

```diff
@@ -238,7 +257,7 @@ public ByteBuf next() throws IOException, InterruptedException {
                     break;
                   }
                 }
-                results.put(Unpooled.wrappedBuffer(buffer));
+                results.put(Pair.of(currentChunkIndex, Unpooled.wrappedBuffer(buffer)));
```

Comment:
A chunk that has been returned does not mean that the chunk is read fully. IMO, I think you should return the chunk index with the chunk and record the chunk index as fully read after the celeborn input stream consumed the chunk fully.